lib/sunrise/sunrise.php

```diff
@@ -67,12 +67,18 @@ function handle_not_found_error( $error_type ) {
 	$is_web_request = Context::is_web_request();
 	if ( $is_web_request ) {
 		$is_maintenance_mode = Context::is_maintenance_mode();
+		$is_overdue_locked   = Context::is_overdue_locked();
 		if ( $is_maintenance_mode ) {
 			// 503 prevents page from being cached.
 			// We handle healthchecks earlier and don't have to worry about them.
 			$status_code = 503;
 			header( 'X-VIP-Go-Maintenance: true' );
 			$error_doc = sprintf( '%s/mu-plugins/errors/site-maintenance.html', WP_CONTENT_DIR );
+		} elseif ( $is_overdue_locked && Context::is_vip_env() ) {
+			// Similar with maintenance mode, but with a different message.
+			$status_code = 402;
+			header( 'X-VIP-402: true' );
+			$error_doc = sprintf( '%s/mu-plugins/errors/site-shutdown.html', WP_CONTENT_DIR );
```

Comment:
Looking at this closer, I wonder if we actually need this code block at all.

The regular locked / not locked check happens separately in 000-vip-init.php ... do we need to handle it again here too? The cross section of sites that are 1) locked out, and 2) have site not found errors is probably pretty tiny, and worst case, the end user is still seeing an error page and not real content.

Just trying to reduce the amount of code and duplication we have, if it makes sense.

Reply:
That's a very good point. I think that I will remove this from the sunrise.php. As you said, worse case scenario, an error will still be displayed instead of any content.